Realm/RLMSyncPermissionResults.mm

```diff
@@ -115,10 +112,8 @@ - (NSUInteger)indexOfObject:(RLMSyncPermission *)object {
     if ([path rangeOfString:@"~"].location != NSNotFound) {
         path = [path stringByReplacingOccurrencesOfString:@"~" withString:object.identity];
     }
-    // Build the predicate. (Don't match the actual permission flags; path and identity identify a permission.)
-    NSPredicate *p = [NSPredicate predicateWithFormat:@"%K = %@ AND %K = %@",
-                      RLMSyncPermissionSortPropertyPath, path,
-                      RLMSyncPermissionSortPropertyUserID, object.identity];
+    // Build the predicate. Check based on the path.
+    NSPredicate *p = [NSPredicate predicateWithFormat:@"%K = %@", RLMSyncPermissionSortPropertyPath, path];
     return [self indexOfObjectWithPredicate:p];
```

Comment:
There isn't really any advantage to building the predicate first over just doing `return [self indexOfObjectWhere:@"%K = %@", RLMSyncPermissionSortPropertyPath, path];`